typescript/cli/src/registry/MergedRegistry.ts

```diff
@@ -0,0 +1,154 @@
+import { Logger } from 'pino';
+
+import {
+  BaseRegistry,
+  ChainAddresses,
+  GithubRegistry,
+  IRegistry,
+  RegistryContent,
+  RegistryType,
+} from '@hyperlane-xyz/registry';
+import { LocalRegistry } from '@hyperlane-xyz/registry/local';
+import {
+  ChainMap,
+  ChainMetadata,
+  ChainName,
+  WarpCoreConfig,
+} from '@hyperlane-xyz/sdk';
+import {
+  isHttpsUrl,
+  objKeys,
+  objMerge,
+  rootLogger,
+} from '@hyperlane-xyz/utils';
+
+export interface MergedRegistryOptions {
+  registryUris: Array<string>;
+  isDryRun?: boolean;
+  logger?: Logger;
+}
+
+export class MergedRegistry extends BaseRegistry implements IRegistry {
+  public readonly type = RegistryType.Local;
+  public readonly registries: Array<IRegistry>;
+  public readonly isDryRun: boolean;
+
+  constructor({ registryUris, logger, isDryRun }: MergedRegistryOptions) {
+    logger ||= rootLogger.child({ module: 'MergedRegistry' });
+    super({ uri: '__merged_registry__', logger });
+
+    if (!registryUris.length)
+      throw new Error('At least one registry URI is required');
+
+    this.registries = registryUris.map((uri, index) => {
+      // If not provided, allow the GithubRegistry to use its default
+      if (isHttpsUrl(uri)) {
+        return new GithubRegistry({ uri, logger: logger!.child({ index }) });
+      } else {
+        return new LocalRegistry({ uri, logger: logger!.child({ index }) });
+      }
+    });
+
+    this.isDryRun = !!isDryRun;
+  }
+
+  async listRegistryContent(): Promise<RegistryContent> {
```

Comment:
nice
might be overkill but you could pull out the `map(registry => r.fn).reduce(objMerge))` into a helper with `fn keyof IRegistry` input